crates/core/tedge_watchdog/src/systemd_watchdog.rs

```diff
@@ -112,13 +114,26 @@ async fn monitor_tedge_service(
 ) -> Result<(), WatchdogError> {
     let client_id: &str = &format!("{}_{}", name, nanoid!());
     let mqtt_config = get_mqtt_config(tedge_config_location, client_id)?
-        .with_subscriptions(res_topic.try_into()?);
+        .with_subscriptions(res_topic.try_into()?)
+        .with_last_will_message(Message {
+            topic: Topic::new_unchecked("tedge/health/tedge-watchdog"),
+            payload: json!({
+                "status": "down",
+                "pid": process::id()})
+            .to_string()
+            .into(),
+            qos: mqtt_channel::QoS::AtLeastOnce,
+            retain: true,
+        });
```

Comment:
This last-will message construction is the same for all the tedge daemons.

The best is to add a `tedge_api::health::get_last_will_message(daemon_name: String)` function.